File: Excel/627 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/627 Alphabets Staircase_2.xlsx"
test = read_excel(path, range = "A2:BA53", col_names = F)

al = LETTERS
M = matrix(NA , nrow = 52,  ncol = 53)

for (i in 1:26) {
  indices = list(c(2*i - 1, 2*i - 1 ), c(2*i - 1, 2*i), c(2*i - 1, 2*i + 1), c(2*i, 2*i + 1))
  for (index in indices) {
    M[index[1], index[2]] = al[i]
  }
}

M = as.data.frame(M)
names(test) = names(M)

all(M == test, na.rm = T)
# [1] TRUE

File: Excel/600-699/627/627 Challenge.py
import pandas as pd
import numpy as np
import string

path = "627 Alphabets Staircase_2.xlsx"
test = pd.read_excel(path, header=None, usecols="A:BA", nrows=52, skiprows=1)

al = list(string.ascii_uppercase)
M = np.full((52, 53), np.nan, dtype=object)

for i in range(1, 27):
    indices = [(2*i - 2, 2*i - 2), (2*i - 2, 2*i - 1), (2*i - 2, 2*i), (2*i - 1, 2*i)]
    for index in indices:
        M[index[0], index[1]] = al[i-1]

M = pd.DataFrame(M)
test.columns = M.columns

print(test.equals(M))
